apps/vaporgui/PTFEditor.cpp

```diff
@@ -134,4 +142,57 @@ void PTFEditor::updateGUI() const
     for (auto a : _histogramActions) a->setEnabled(_histogram->IsShown());
 }
 
+void PTFEditor::Update(VAPoR::ParamsBase *params, VAPoR::ParamsMgr *paramsMgr, VAPoR::DataMgr *dataMgr) {
+    PWidget::Update(params, paramsMgr, dataMgr);
+    if (_expandable==true) {
+        std::string name, inst;
+        getExpandedPTFEditorInfo(name, inst);
+
+        if (_expandedPTFEditor!=nullptr) {
+            _expandedPTFEditor->setWindowTitle(QString::fromStdString(name));
+            _expandedPTFEditor->Update(params, paramsMgr, dataMgr);
+        }
+    }
+}
+
+void PTFEditor::getExpandedPTFEditorInfo(std::string &name, std::string& type) {
+    ParamsMgr* pm = getParamsMgr();
+    GUIStateParams* p = dynamic_cast<GUIStateParams *>(pm->GetParams(GUIStateParams::GetClassType()));
+    type = p->GetActiveRendererInst();
+    p->GetActiveRenderer(p->GetActiveVizName(), type, name);
+    if (dynamic_cast<PColormapTFEditor*>(this)) {
+        name+="_Colormap";
+        type+="_Colormap";
+    }
+}
+
+void PTFEditor::showExpandedPTFEditor() {
+    if (_expandedPTFEditor==nullptr) {
+        _expandedPTFEditor = new PTFEditor(_tag, _elements, _label, false);
+        connect(_expandedPTFEditor, SIGNAL(closed()), this, SLOT(closeExpandedPTFEditor()));
+
+        if (_showColormapBasedOnParam==true) _expandedPTFEditor->ShowColormapBasedOnParam(_showColormapBasedOnParamTag, _showColormapBasedOnParamValue);
+        if (_showOpacityBasedOnParam==true) _expandedPTFEditor->ShowOpacityBasedOnParam(_showOpacityBasedOnParamTag, _showOpacityBasedOnParamValue);
+
+        _expandedPTFEditor->setAttribute(Qt::WA_ShowWithoutActivating);
+        _expandedPTFEditor->setAttribute(Qt::WA_DeleteOnClose);
+    }
+    _expandedPTFEditor->raise();
+
+    // It was originally thought to store the open expanded PTFEditors in GUIStateParams.
+    // This lead to problems and the team agreed that it's not important to restore them if
+    // they were expanded in a session, so we manually call Update here.
+    Update(getParams(), getParamsMgr(), getDataMgr());
+}
+
+void PTFEditor::closeExpandedPTFEditor() {
+    _expandedPTFEditor->close();
+    _expandedPTFEditor=nullptr;
+}
+
+void PTFEditor::closeEvent(QCloseEvent* event) {
+    emit closed();
```

Comment:
Why not simply call `closeExpandedPTFEditor`

Reply:
We need to specifically capture the expanded PTFEditor's closeEvent so the containing PTFEditor can then set the expanded to nullptr.

If we don't, then the checks for _expandedPTFEditor==nullptr will fail, particularly in PTFEditor::showExpandedPTFEditor().

Reply:
So you're saying when PTFEditor receives its own `close` event in PTFEditor::closeEvent, it needs to then re-emit different `closed` event which it can then capture again in PTFEditor::closeExpandedPTFEditor where it tells the expanded window to close?

Reply:
No, the PTFEditor is not receiving its own close event.  When the PTFEditor's _expanded instance_ closes, the containing PTFEditor must to set the _expanded instance_ to nullptr.  See the connection being made at PTFEditor.cpp:175.

Reply:
I see. There are two instances of the PTFEditor with slightly different behavior sending signals to each other. Could the window container behavior be extracted into its own class?

Reply:
I've extracted the signal into a new subclass, ExpandedPTFEditor.